cocotb/utils.py

```diff
@@ -619,5 +619,11 @@ def __neg__(self):
     def __abs__(self):
         return SimTime(abs(self.steps))
 
+    def __str__(self):
+        return repr(self)
+
     def __repr__(self):
-        return "{}({})".format(self.__class__.__name__, self.steps)
+        for unit in ('fs', 'ps', 'ns', 'us', 'ms', 'sec'):
+            time = get_time_from_sim_steps(self.steps, unit)
+            if 0 <= time and time < 1000:
+                return "{}({}, '{}')".format(self.__class__.__name__, time, unit)
```

Comment:
Nit:`"'{}'".format(s)` is in general better spelt `"{!r}".format(s)` (r for repr), which handles embedded quotes and escape sequences for you. Works either way here though.